client/my-sites/plan-features-2023-grid/hooks/npm-ready/data-store/use-grid-plans.tsx

```diff
@@ -129,14 +129,23 @@ interface Props {
 		[ key: string ]: boolean;
 	};
 	showLegacyStorageFeature?: boolean;
+
+	/**
+	 * If the subdomain generation is unsuccessful we do not show the free plan
+	 */
+	isSubdomainGenerated: boolean;
```

Comment:
This feels indirect to me and makes potential reuse of this hook outside of admin/Calypso tricky i.e. to show free plan now you need to force this to `true` irrespective of needing these subdomains or not. 

Does this make sense? Maybe try its inverse and also set it to optional. This prop **should be** optional IMO.

Reply:
Good catch @chriskmnds 👍 I agree with this observation